09-big-three/SOLUTION.md

```diff
@@ -84,6 +84,36 @@ We could optimize the placement logic by utilizing a binary search sort of techn
 
 With this, we have a linear solution and we only had to make one pass.
 
+
+## Solution 3: The triple variable
+
+This solution sets variables for the three highest values, and updates them
+as a new high value is encountered while looping through the values. This reduces actual runtime by eliminating the 'while' loop for each value in the array.
+
+```js
+const getBigThree = (numsArr) => {
+  let highest = 0,
+    nextHighest = 0,
+    thirdHighest = 0;
+
+  numsArr.forEach(num => {
+    if(num >= highest) {
+      thirdHighest = nextHighest;
+      nextHighest = highest;
+      highest = num;
+    } else if(num >= nextHighest) {
+      thirdHighest = nextHighest;
+      nextHighest = num;
+    } else if(num > thirdHighest) {
+      thirdHighest = num;
+    }
+  })
+  return [thirdHighest, nextHighest, highest];
+}
```

Comment:
@mwhitman189 this is really cool. How about instantiating an array of 3 null values and then filling in instead?

```suggestion
const getBigThree = (array) => {
  let largest = [null, null, null];
  
  for (const num of array) {
    if(num > largest[2] || !largest[2]) {
      largest[0] = largest[1];
      largest[1] = largest[2];
      largest[2] = num;
    } else if(num > largest[1] || !largest[1]) {
      largest[0] = largest[1];
      largest[1] = num;
    } else if(num > largest[0] || !largest[0]) {
      largest[0] = num;
    }
  }
  
  return largest;
}
```

Reply:
Great suggestion, @shin10kudev ! Thanks